pennylane_lightning_gpu/src/tests/Test_Generators.cpp

```diff
@@ -0,0 +1,1130 @@
+#include <cmath>
+#include <complex>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "GateGenerators.hpp"
+#include "StateVectorCudaManaged.hpp"
+#include "TestHelpers.hpp"
+
+#ifndef _USE_MATH_DEFINES
+#define _USE_MATH_DEFINES
+#endif
+
+using namespace Pennylane::CUDA;
+using namespace Pennylane::CUDA::Generators;
+using namespace Pennylane::Algorithms;
+
+// NOTE: the scaling factors are implicitly included in the Adjoint Jacobian
+// evaluation, so excluded from the matrices here.
+
+TEST_CASE("Generators::applyGeneratorRX_GPU", "[GateGenerators]") {
+    // grad(RX) = grad(e^{-i*0.5*PauliX*a}) => -i*0.5*PauliX
+    std::vector<typename StateVectorCudaManaged<double>::CFP_t> matrix{
+        cuGates::getPauliX<typename StateVectorCudaManaged<double>::CFP_t>()};
+
+    for (std::size_t num_qubits = 1; num_qubits <= 5; num_qubits++) {
+        for (std::size_t applied_qubit = 0; applied_qubit < num_qubits;
+             applied_qubit++) {
+            SVDataGPU<double> psi(num_qubits);
+            SVDataGPU<double> psi_direct(num_qubits);
```

Comment:
This will create a |0...0> state, right? If I'm reading this correctly, the double loop will make sure this is apply to at least one non-trivial case (albeit a special one), but most of them are just mixing zeros. The last is also a special case worth testing I would say, making sure (0 +0 == 0), but do you think it is worth writing functions randomly initializing the state and test on that?

Reply:
I agree stochasticity is to be avoided as possible in tests. Here I would say it is not too bad, since where are checking one way to apply an operation against another. 

Thinking about it, we might not even need random values to address the point. We could just generate a normalized vector (1..2^n). Would that do?